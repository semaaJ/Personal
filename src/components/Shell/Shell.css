* {
	-webkit-transition-property: all;
	-webkit-transition-duration: .2s;
  -moz-transition-timing-function: cubic-bezier(100,50,21,6);
	-moz-transition-property: all;
  -moz-transition-timing-function: cubic-bezier(100,50,21,6);
}

.shell-wrap {
  display: flex;
  flex-direction: column;
  width: 900px;
  margin: 0 auto 0 auto;
  margin-bottom: 20px;
  box-shadow: 0 0 30px rgba(0,0,0,0.4);
}

.shell-top-bar {
  text-align: center;
  padding: 8px 0;
  text-shadow: 1px 1px 0 rgba(255,255,255,0.5);
  font-size: 0.85em;
  background: #A0A0A0;
}

.shell-body {
  margin: 0;
  height: 450px;
  padding: 5px;
  display: flex;
  flex-direction: column;
  background: #383737;
  color: #fff;
  font: 0.8em 'Karla', Consolas, 'Courier New';
  line-height: 1.6em;
  text-align: left;
  border-bottom-right-radius: 3px;
  border-bottom-left-radius: 3px;
}

.skip {
  display: flex;
  justify-content: right;
  cursor: pointer;
}

.skip:hover {
  color: #00EEEE;
  transition: 1s ease;
}

.staticShellContainer {
  margin-top: 20px;
}

.object {
  font: 14px 'Inconsolata', monospace;
  margin-left: 30px;
  margin-right: 50px;
}

.function {
  font-weight: 400;
  font: 14px 'Inconsolata', monospace;
  margin-left: 30px;
  margin-right: 50px;
  padding: 6px 0 2px 0;
}

.return {
  font: 14px 'Inconsolata', monospace;
  font-weight: 400;
  margin-left: 30px;
  margin-right: 50px;
  padding: 6px 0 2px 0;
}

.terminalName {
  display: inline;
  color: #00FF00;
}

.tilda {
  display: inline;
  color: #FF69B4;
}

.highlight {
  display: inline;
  color: #00EEEE;
}

.link {
  cursor: pointer;
  color: #00EEEE;
}

.link:hover {
  color: #00EEEE;
}