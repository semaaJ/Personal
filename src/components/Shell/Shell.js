import React, { useState } from 'react';
import TypeIt from 'typeit-react';
import Icon from '@mdi/react'
import { mdiSkipForward } from '@mdi/js'
import Draggable from 'react-draggable';
import './Shell.css';

const shellData = (bool) => {
    const cssClass = bool ? "staticShellContainer" : "";
    return (
        <div className={cssClass}>
            <div className="object"><div className="terminalName">james@personal</div>:<div className="tilda">~</div><div className="highlight">$</div> cat james.txt</div>
            <p className="function">
                Hi! I'm James, a 22 year old Junior Software Engineer from <i className="highlight">Ireland</i>. 
                I have just completed my final year in <i className="highlight">Dublin City Univeristy</i>, where I studied Computer Applications & Software Engineering. 
            </p>

            <p className="return">
                I'm a creative, hardworking and passionate developer who enjoys building the complexities of back-end applications and hiding them behind beautiful web applications. 
            </p> 
            
            <p className="function">
                I have a strong understanding of <i className="highlight">JavaScript (ES6)</i>, <i className="highlight">Python</i>, <i className="highlight">HTML/CSS</i> and am familiar with <i className="highlight">Java</i> and
                have a deep love for the <i className="highlight">React</i> framework. I enjoy the translation of UI designs into modular and highly testable components.
            </p>

            <p className="return">
                My CV is availible <a href="https://www.google.com" className="link">here</a> and if you would like to get in touch with me, 
                feel free to contact me via <a href="https://www.google.com" className="link">email</a>, <a href="https://www.google.com" className="link">Github</a> or <a href="https://www.google.com" className="link">LinkedIn</a>!
            </p> 

            <p className="function">
                When I'm not pushing pixels around I'm probably reading, gaming, listening to podcasts/music or enjoying some brews with friends! 
            </p> 
        </div>
    )
}

const Shell = (props) => {
    const { topWindow, setTopWindow } = props;
    const [skip, setSkip] = useState(false);
    
    const isTopWindow = topWindow === "Shell" ? "shell-wrap topIndex" : "shell-wrap";

    const onSkipHandler = () => {
        setSkip(true);
    }
    
    return (
        <Draggable>
            <div className={isTopWindow} onClick={() => setTopWindow("Shell")}>
                <div className="shell-top-bar" />
                <div className="shell-body">
                    <div className="skip">
                        <Icon 
                            className="icon"
                            onClick={onSkipHandler}
                            path={mdiSkipForward} 
                            size={1}
                        />
                    </div>
                        { skip ? shellData(true) : 
                            <TypeIt
                                options={{
                                    speed: 10,
                                    waitUntilVisible: true,
                                }}
                            >
                            { shellData() }
                        </TypeIt>
                        }
                </div>
            </div>
        </Draggable>
    )
}   

export default Shell;