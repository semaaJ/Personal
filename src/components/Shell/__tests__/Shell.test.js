import React from 'react';
import Shell from '../Shell';
import { mount } from 'enzyme';
import '../../../__mocks__/intersectionObserverMock';
import "../../../setupTests"

test('should render Shell correctly', () => {
    const component = mount(<Shell />);
    expect(component.html()).toMatchSnapshot();
});

