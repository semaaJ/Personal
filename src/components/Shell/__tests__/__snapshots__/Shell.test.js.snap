// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Shell correctly 1`] = `"<div class=\\"shell-wrap\\"><div class=\\"shell-top-bar\\"></div><div class=\\"shell-body\\"><div class=\\"skip\\"><svg viewBox=\\"0 0 24 24\\" style=\\"width: 1.5rem; height: 1.5rem;\\" role=\\"presentation\\" class=\\"icon\\"><path d=\\"M4,5V19L11,12M18,5V19H20V5M11,5V19L18,12\\" style=\\"fill: currentColor;\\"></path></svg></div><div style=\\"opacity: 1;\\"><span data-typeit-id=\\"4912939\\"><span class=\\"ti-cursor\\" style=\\"display: inline;\\">|</span></span></div></div></div>"`;
